app/guides/utils.ts

```diff
@@ -0,0 +1,96 @@
+"use server";
+import { ReturnDocument } from "../models/return";
+import {
+  FeedbackDocument,
+  ReviewedFeedbackDocument,
+  Vote,
+} from "../models/review";
+import {
+  ReturnStatus,
+  FeedbackStatus,
+  ReviewsReceivedStatus,
+  ReviewsGivenStatus,
+} from "./types";
+
+export const calculateReturnStatus = (
+  returnedGuides: ReturnDocument[],
+  feedbackReceived: FeedbackDocument[]
+): ReturnStatus => {
+  if (returnedGuides.length === 0) {
+    return ReturnStatus.NOT_RETURNED;
+  }
+
+  const latestSubmission = returnedGuides.reduce(
+    (latestSubmission, submission) => {
+      if (submission.createdAt > latestSubmission.createdAt) {
+        return submission;
+      }
+      return latestSubmission;
+    }
+  );
+  const feedbackOnLatestSubmission = feedbackReceived.filter((feedback) => {
+    return feedback.return === latestSubmission._id;
+  });
+
+  if (
+    feedbackOnLatestSubmission.filter(
+      (feedback) => feedback.vote === Vote.NO_PASS
```

Comment:
We should probably change this because we want the red X only to appear if two reviewers give a no pass. Although it should be ok to have a red X if you only have one review and it is a no pass...

Reply:
Changed to > 1